power_grid_model_c/power_grid_model/include/power_grid_model/main_model_impl.hpp

```diff
@@ -445,15 +442,7 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
     }
     // get sequence idx map of an entire batch for fast caching of component sequences
     SequenceIdx get_sequence_idx_map(ConstDataset const& update_data, ComponentFlags const& to_store) const {
-        // TODO: (jguo) this function could be encapsulated in UpdateCompIndependence in update.hpp
-        return run_functor_with_all_types_return_array([this, &update_data, &to_store]<typename CT>() {
-            if (!to_store[index_of_component<CT>]) {
-                return std::vector<Idx2D>{};
-            }
-            auto const independence = check_components_independence<CT>(update_data);
-            validate_update_data_independence(independence);
-            return get_component_sequence<CT>(update_data, 0, independence);
-        });
+        return get_sequence_idx_map(update_data, 0, to_store);
```

Comment:
This is basically for permanent updates, right? Or for scenario 0 only. If so, shouldn't we edit as well the comment explaining the function?

Reply:
no, this is the entry point used when caching the entire batch after determining which component types are independent. the one below is for permanent updates.

will remove this as indeed it is very unclear as to what happens